src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -288,6 +289,11 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                 }
 
                 services.AddSingleton<IHostedService, WorkerConsoleLogService>();
+
+                if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(EnvironmentSettingNames.PodName)))
```

Comment:
use `IsKubernetesManagedHosting` ? if not add a helper in environment extensions